### PR TITLE
Make TQDM work nicely in log streams

### PR DESCRIPTION
**Pull request by farizrahman4u**
Test simple progress bar:

```python

from tqdm import tqdm
import time
import wandb

wandb.init(project='test_tqdm')

start_epoch = 0
num_epochs = 10
for epoch in tqdm(range(start_epoch, start_epoch+num_epochs), "epochs"):
    for batch in range(100):
        wandb.log({"epoch": epoch, "batch": batch})
        time.sleep(0.01)
```

Test nested progress bar:

```python

from tqdm import tqdm
import time
import wandb

wandb.init(project='test_tqdm_nested')

start_epoch = 0
num_epochs = 10
for epoch in tqdm(range(start_epoch, start_epoch+num_epochs), "epochs"):
    for batch in enumerate(tqdm(range(100), f"{epoch}=")):
        wandb.log({"epoch": epoch, "batch": batch})
        time.sleep(0.01)
```

### CODE REVIEW

File: wandb/internal/file_stream.py [hunk 1]
```diff
@@ -73,19 +75,25 @@ class CRDedupeFilePolicy(DefaultFilePolicy):
...
+            for line in lines:
+                line = line.split('\r')[-1]
+                if line:
+                    if line.endswith('\x1b\x5b\x41'):
+                        if flag:
+                            ret.pop()
+                            flag = False
+                    else:
+                        ret.append(line + os.linesep)
+                        flag = True
```
Comment (vanpelt) [inline wandb/internal/file_stream.py:91]: What does `flag` actually mean here?  Can we name this variable more descriptively.  A comment above the `endswith` escape sequence would be helpful too.

File: wandb/internal/file_stream.py [hunk 2]
```diff
@@ -73,19 +75,25 @@ class CRDedupeFilePolicy(DefaultFilePolicy):
...
         chunk_id = self._chunk_id
-        self._chunk_id += len(content)
-        if content and content[-1].endswith('\r'):
-            self._chunk_id -= 1
+        self._chunk_id += len(ret)
```
Comment (vanpelt) [inline wandb/internal/file_stream.py:93]: This seems wrong.  There are other libraries besides tqdm that just keep writing the same line with a carriage return to indicate progress.  That why we used to just overwrite the previous chunk in this case.  With this new logic I think we'll stream all the lines.  I would try something like the following:

```
spinner_states = ['-', '\\', '|', '/']
for s in spinner_states:
    sys.stdout.write(s)
    sys.stdout.flush()
print("")
```

And ensure that only sends one line to wandb.

Comment (farizrahman4u) [reply wandb/internal/file_stream.py:96]: Only 1 line gets sent:

```python
spinner_states =['-', '\\', '|', '/']
for s in spinner_states * 10:
    sys.stdout.write('\r')
    sys.stdout.write(s)
    sys.stdout.flush()
    time.sleep(0.05)
print("")

```

Comment (vanpelt) [reply wandb/internal/file_stream.py:96]: 👍

**Pull request merged**